content/blog/pair-programming-101/index.md

```diff
@@ -0,0 +1,76 @@
+---
+title: Pair Programming 101
+date: "2021-09-23T11:11:00.284Z"
+description: "How to implement productive pair programming sessions?"
+---
+
+Pair Programming is easily one of my favourite activities on my day to day job, but getting them right is not always easy. The idea that it's enough to set up those sessions and everything will flow permeates our industry, but things rarely go smoothly without prior planning. Below is a non-exhaustive documentation of things that have worked for me in the past.
+
+Before we start it's worth mentioning that as people get to know and trust each other it becomes easier to have impromptu pair programming sessions.
+
+## What is the goal of a pair programming session?
+
+Let's investigate which goals are best suited for pair programming sessions. This list is non-exhaustive, but those three have been my top goals for some years now.
+
+#### Sharing knowledge
+
+Pair programming is a great tool to onboard new peers into a codebase, project or even a single task. The idea of being able to ask questions while diving into critical parts of the code from someone with higher context makes this goal a particularly good fit for pair programming.
+
+#### Brainstorming
+
+It's very common for developers to get stuck trying to come up with an implementation that they consider solid. Pair programming gives them the opportunity to brainstorm with different developers and gather insights that might unlock a great solution for the problem at hand.
+
```

Comment:
very interesting!

This also might go into this category of Brainstorming: Pairing is also great for a very short feedback loop. You can get your "review" right away without investing a lot of time back and forth for the review.


Also, another thing that pairing is really helpful for is to see how other people navigate the code. There are a ton of tips and tricks we can learn from people who dealt with various IDEs and their customizations. Also, a lot of time devs automate mundane tasks that can inspire us. Watching someone do their thing widens our perspective and makes us better and more efficient devs.